source/info.d

```diff
@@ -13,182 +21,69 @@ struct Exercise
-            function and change the code to `assert(isNan(f));`".outdent.strip(),
-        type: compile,
-    },
-
-    {
-        name: "floating_point4",
-        path: buildPath("exercises", "floating_point", "floating_point4.d"),
-        hint: "No hints this time ;)",
-        type: compile,
-    },
-
-// functions:
-    {
-        name: "functions1",
-        path: buildPath("exercises", "functions", "functions1.d"),
-        hint: "Insert `a + b` after `return`.",
-        type: compile,
-    },
-
-    {
-        name: "functions2",
-        path: buildPath("exercises", "functions", "functions2.d"),
-        hint: "Insert the parameter `float a`, e.g `float half(float a)`",
-        type: compile,
-    },
-
-    {
-        name: "functions3",
-        path: buildPath("exercises", "functions", "functions3.d"),
-        hint: "Insert the parameter `float a`, e.g `float half(float a)`",
-        type: compile,
-    },
-
-    {
-        name: "functions4",
-        path: buildPath("exercises", "functions", "functions4.d"),
-        hint: "Create another function 'add' that receives 'float' types.",
-        type: compile,
-    },
-
-    {
-        name: "functions5",
-        path: buildPath("exercises", "functions", "functions5.d"),
-        hint: "
-            Modify the function signature to work with templated types.
-            `T multiply(T)(const T left, const T right)`".outdent.strip(),
-        type: compile,
-    },
-
-// if:
-    {
-        name: "if1",
-        path: buildPath("exercises", "if", "if1.d"),
-        hint: "
-            Use an 'if' statement:
-            ---
-            if (<condition>)
-            {
-                return <identifier>;
-            }
-            else
-            {
-                return <identifier>;
-            }
-            ---
-
-            Note: The parentheses are optional in D for the `if` statement.
-            Note: without parenthesis only the first statement after `if` is
-            evaluated!".outdent.strip(),
-        type: test,
-    },
-
-    {
-        name: "if2",
-        path: buildPath("exercises", "if", "if2.d"),
-        hint: "No hints this time ;)".outdent.strip(),
-        type: test,
-    },
-
-    {
-        name: "if3",
-        path: buildPath("exercises", "if", "if3.d"),
-        hint: "Use the ternary operator, e.g `return a > b ? a : b;`".outdent.strip(),
-        type: test,
-    },
-
-// quiz
-    {
-        name: "quiz1",
-        path: buildPath("exercises", "quiz", "quiz1.d"),
-        hint: "No hints this time ;)".outdent.strip(),
-        type: test,
-    },
-];
+/**
+Reads TOML `filepath` and parses its contents into `Exercise`s.
+
+Params:
+	filepath = '.toml' file to parse
+
+Returns: an `immutable Exercise[]` with the parsed exercise's data.
+*/
+immutable(Exercise[]) exercisesFromTOML(in string filepath)
+	in (filepath.exists)
+{
+	debug infof("parsing file '%s'", filepath);
```

Comment:
You probably want a debug log instead of using `debug` blocks.

Reply:
isn't this the same thing?